codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -306,6 +307,79 @@ private void generateOperationSerializerMiddleware(GenerationContext context, Op
         });
     }
 
+    // Generates operation deserializer middleware that delegates to appropriate deserializers for the error,
+    // output shapes for the operation.
+    private void generateOperationDeserializerMiddleware(GenerationContext context, OperationShape operation) {
+        GoStackStepMiddlewareGenerator middleware = GoStackStepMiddlewareGenerator.createDeserializeStepMiddleware(
+                ProtocolGenerator.getDeserializeMiddlewareName(operation.getId(), getProtocolName()));
+
+        SymbolProvider symbolProvider = context.getSymbolProvider();
+        Model model = context.getModel();
+
+        ApplicationProtocol applicationProtocol = getApplicationProtocol();
+        Symbol responseType = applicationProtocol.getResponseType();
+        GoWriter goWriter = context.getWriter();
+
+        middleware.writeMiddleware(goWriter, (generator, writer) -> {
+            writer.addUseImports(GoDependency.FMT);
+
+            writer.write("out, metadata, err = next.$L(ctx, in)", generator.getHandleMethodName());
+            writer.write("if err != nil { return out, metadata, err }");
+            writer.write("");
+
+            writer.write("response, ok := out.RawResponse.($P)", responseType);
+            writer.openBlock("if !ok {", "}", () -> {
+               writer.write(String.format("return out, metadata, &aws.DeserializationError{Err: %s}",
```

Comment:
We really need to break this dependency. Can you move this type into the base repo?

Reply:
Considering this as an incremental change. Will coordinate with how serializers handle this, and create a separate PR accordingly